src/v/random/generators.cc

```diff
@@ -21,13 +21,18 @@ void fill_buffer_randomchars(char* start, size_t amount) {
 static constexpr std::string_view chars
   = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789";
 
-ss::sstring gen_alphanum_string(size_t n) {
+ss::sstring gen_alphanum_string(size_t n, bool use_secure_rng) {
     // do not include \0
     static constexpr std::size_t max_index = chars.size() - 2;
     std::uniform_int_distribution<size_t> dist(0, max_index);
     auto s = ss::uninitialized_string(n);
-    std::generate_n(
-      s.begin(), n, [&dist] { return chars[dist(internal::gen)]; });
+    std::generate_n(s.begin(), n, [&dist, use_secure_rng] {
+        if (use_secure_rng) {
+            return chars[dist(internal::secure_public_rng)];
+        } else {
+            return chars[dist(internal::gen)];
+        }
+    });
```

Comment:
Hopefully the optimiser can see that the branch will always be the same.

Reply:
Sorry I'm not following what you mean :)

Reply:
The `if (use_secure_rng)` branch is inside the loop. It's pretty minor, probably not worth hoisting.

Reply:
ah ok now I understand what you mean :)